File: parsing/src/test/resources/templates/c_freemarker/ComplexTemplate.c
#include <stdio.h>

int test1(int i);
void test2(char* str) {
    printf("%s", str);
}

int main() {
    int i;
    int j;
    char* ${str} = "hello world";

    j = test1(j);

    char* foo = "foo";
    test2(foo);

    <#if anything>
    int foo = 42;
    <#else>
    int foo = 84;
    </#if>

    return 0;
}

int test1(int i) {
    return i + 42;
}

File: aim-pattern-detection/src/test/resources/templates/java_freemarker/ComplexTemplate.java
package a.b.c.d;

import abc.def.ghi.jkl;
import xyz.uvw.rst;
import foo.bar.foobar;

public class ${className} {

	private int i;
	private int j;
	private String ${str} = "hello world";

	public ${className}(){

	}

	public void test1() {
		int foo = 1;
		foo = foo + 42;
		System.out.println(foo);
	}

	public int test2() {
		int bar = 1;
		foo = foo + 42;
		return foo;
	}

	<#if getter>
	public int getI() {
		return this.i;
	}
	</#if>

	<#if anything>
	public String toString() {
		return "hello";
	}
	<#else>
	public String toString() {
		return "world";
	}
	</#if>
}